.github/workflows/build_test_containers.yml

```diff
@@ -0,0 +1,95 @@
+name: Build, test and publish containers
+
+on:
+  pull_request:
+    branches:
+      -  '*'
+  push:
+    branches:
+      -  '*'
+    tags:
+      - '*'
+
+jobs:
+
+  simulation-tests:
+    name: Build containers and run boptest simulation tests
+    runs-on: ubuntu-22.04
+    timeout-minutes: 240
+    env:
+      COMPOSE_PROJECT_NAME: boptest_service
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+
+      - name: Build and run stack
+        run: |
+          cd service
+          docker compose build
+          docker compose up -d web worker
+
+      - name: dump docker logs
+        uses: jwalton/gh-docker-logs@v1
+
+      - name: Upload test cases to minio
+        run: |
+          # The provision script will wait for the web service to be available,
+          # so there is no need for an external wait-for-it script
+          cd service
+          docker compose run --no-deps provision
+          curl http://localhost/testcases
+
+      - name: Run example
+        run: |
+          python --version
+          pip install requests matplotlib numpy pandas
+          PYTHONPATH=$GITHUB_WORKSPACE
+          echo "PYTHONPATH=$PYTHONPATH" >> $GITHUB_ENV
+          # Storing PYTHONPATH above doesn't work for python so setting it below at run 
+          PYTHONPATH=$PWD python examples/python/testcase1.py 
+
+      - name: Run tests
+        run: |
+           cd service
+           docker compose run --no-deps test
+
+      - name: Dump docker logs on failure
+        if: failure()
+        uses: jwalton/gh-docker-logs@v1
+
+#-------------------------  Push to GitHub registry (disabled) -----------------------------
```

Comment:
This line says "disabled" but the code below is not commented out.  Double check if it should be disabled or adjust comment.

Reply:
@dhblum  To answer you question, yes, the GitHub Action runner currently will only publish images when there is a push to either develop, experimental and tags. However, the CI will run on any commit for all branches and run the test cases, it just won't publish the images.  

Also, I removed that comment line that said "disabled"